rocketpy/Dispersion.py

```diff
@@ -87,841 +101,209 @@ def __init__(
+            self.import_inputs()
+        except:
+            self._dispersion_input_file = ""
 
-        # Rail button
-        dictionary = self.__process_class_from_dict(
-            dictionary,
-            self.rocket,
-            "railbuttons",
-            "Rocket with Rail Buttons",
-            "RailButtons",
-        )
+        try:
+            self.import_outputs()
+        except:
+            self._dispersion_output_file = ""
 
-        # Flight
-        dictionary = self.__process_class_from_dict(
-            dictionary, self.flight, "flight", "Flight"
-        )
+        try:
+            self.import_errors()
+        except:
+            self._dispersion_error_file = ""
 
-        # Parachute and Aero Surfaces need special treatment
-        # Parachute
-        dictionary = self.__process_parachute_from_dict(dictionary)
+        # TODO: Initialize variables so they can be accessed by MATLAB
 
-        # Aerodynamic Surfaces
-        dictionary = self.__process_aerodynamic_surfaces_from_dict(dictionary)
+        return None
 
-        return dictionary
+    # getters and setters for dispersion input/output/error files
+    @property
+    def dispersion_input_file(self):
+        # try is for when the file has not been opened yet
+        try:
+            # Resets cursor position to the beginning of the file
+            self._dispersion_input_file.seek(0)
+            return self._dispersion_input_file
+        except:
+            return self._dispersion_input_file
+
+    @dispersion_input_file.setter
+    def dispersion_input_file(self, value):
+        self._dispersion_input_file = value
+        return self._dispersion_input_file
+
+    @property
+    def dispersion_output_file(self):
+        # try is for when the file has not been opened yet
+        try:
+            # Resets cursor position to the beginning of the file
+            self._dispersion_output_file.seek(0)
+            return self._dispersion_output_file
+        except:
+            return self._dispersion_output_file
+
+    @dispersion_output_file.setter
+    def dispersion_output_file(self, value):
+        self._dispersion_output_file = value
+        return self._dispersion_output_file
+
+    @property
+    def dispersion_error_file(self):
+        # try is for when the file has not been opened yet
+        try:
+            # Resets cursor position to the beginning of the file
+            self._dispersion_error_file.seek(0)
+            return self._dispersion_error_file
+        except:
+            return self._dispersion_error_file
+
+    @dispersion_error_file.setter
+    def dispersion_error_file(self, value):
+        self._dispersion_error_file = value
+        return self._dispersion_error_file
+
+    # cached properties
+    @cached_property
+    def inputs_log(self):
+        """Save inputs_log log from a file into an attribute for easy access"""
+        # TODO: this currently does not work because of parachute trigger function
```

Comment:
Does it still not work? It seems that one of the last commits solved it.

Reply:
Yep it is working. Just forgot to remove the Yep it is working. Just forgot to remove the TODO

What I did was I made the Functions be saved just as their `__repr__` ( the `Function R1 -> R1` thingy) and the trigger function just saves as a string too